### PR TITLE
Reflavors IAA, removes certain items from IAA uplinks, adds TC stealing functionality to uplinks

### PR DESCRIPTION
**Pull request by Sishen1542**
## About The Pull Request
Textual reflavoring of IAA, little memey but it's fun. 
- Addition of Krav Maga to IAA uplinks.
- TC theft: to take TC from a hidden uplink, all you have to do is slap the item (pda, pen, headset) to your currently active uplink. No visible messages except to the user.

Removal of a lot of items from IAA uplinks, such as:
- Northstar gloves
- Random Bundles
- Surplus Crates
- Random Items
- Revolvers
- Dswords, Eswords and Rapier
- Holopara
- Power Fist
- X-4, Syndiebombs & Minibombs
- Blast cannon, Clockwork & Bloodcult items

> _**If you have anything to suggest adding/removing from IAA uplinks, leave a comment and the rationale behind your thoughts.**_

## Why It's Good For The Game
IAA is a fun mode that has been neglected. It's very hectic for those involved, as there will always be someone hunting you. By reflavoring some texts I hope to reduce the abuse of cooperating with security, while by limiting the loud/obvious items in the uplinks I hope to reduce the heat that agents get.

I initially planned for IAA to get additional TC on target kill, but I figured going outside the box would add an interesting dynamic to other roundtypes too.

## Changelog
:cl:
eventually
/:cl:

### CODE REVIEW

**Comment by timothyteakettle** [comment]
make sure you can't get the detomatix as people just go down the list clicking send virus on people which is loud and not how IAA is supposd to go

**Comment by Hatterhat** [comment]
fuck great leader all my homies hate great leader

**Comment by Sishen1542** [comment]
> make sure you can't get the detomatix as people just go down the list clicking send virus on people which is loud and not how IAA is supposd to go

the thought behind allowing detomatix is that it's an easy take down for your target, i would hope that in the mode where you're supposed to be stealthy (save for the scant few syndicate IAA) that they wouldn't be popping them off at any non-targets

**Comment by lolman360** [comment]
add tubarão it'll be funny i swear

also the temblor/black hand

**Comment by lolman360** [comment]
ill probably add rigs purely so we can have powersuits, who doesn't want to pilot a mog around

also imagine the sledger w/ like, a bishop or otomo in ss13

File: code/modules/uplink/uplink_items/uplink_stealth.dm [hunk 1]
```diff
@@ -65,6 +65,16 @@
 	surplus = 0
 	exclude_modes = list(/datum/game_mode/nuclear, /datum/game_mode/nuclear/clown_ops)
 
+/datum/uplink_item/stealthy_weapons/martialartsthree
+	name = "Krav Maga Scroll"
+	desc = "This scroll contains the secrets of an ancient martial arts technique. You will gain special unarmed attacks for \
+			stealthy takedowns."
+	item = /obj/item/book/granter/martial/krav_maga
+	cost = 16
+	player_minimum = 30
```
Comment (necromanceranne) [inline code/modules/uplink/uplink_items/uplink_stealth.dm]: Not sure what the player limit is on IAA, but given you have no super advantageous invisible defenses, beyond additional forms of attack, I don't think you'd need to add a player limit, unlike scarp. It is possible to take down someone with krav maga on your own.

Price seems fine though.

Comment (Sishen1542) [reply code/modules/uplink/uplink_items/uplink_stealth.dm]: IAA player req is 25 which is honestly close enough, so it's possible to remove the minimum for krav.

File: code/datums/components/uplink.dm [hunk 2]
```diff
@@ -102,6 +102,12 @@ GLOBAL_LIST_EMPTY(uplinks)
 		return	//no hitting everyone/everything just to try to slot tcs in!
 	if(istype(I, /obj/item/stack/telecrystal))
 		LoadTC(user, I)
+	if(active)
+		var/datum/component/uplink/hidden_uplink = I.GetComponent(/datum/component/uplink)
+		var/amt = hidden_uplink.telecrystals
```
Comment (silicons) [inline code/datums/components/uplink.dm]: what happens if there's no uplink component? make a null check

File: code/game/objects/items/granters.dm [hunk 3]
```diff
@@ -477,6 +477,23 @@
...
+	name = "parchment scroll"
+	martialname = "krav maga"
+	desc = "A worn parchment scrap written in an ancient language. Somehow you can still understand the lessons!"
+	greet = "<span class='sciradio'>You have learned the ancient martial art of Krav Maga. You have special attacks with which to take down your foes.</span>"
+	icon = 'icons/obj/wizard.dmi'
+	icon_state ="scroll2"
+	remarks = list("Sweep the legs...", "Chop the throat...", "Punch the lungs...", "Get the gold...", "Where are my sick gloves..?")
+
+/obj/item/book/granter/martial/krav_maga/onlearned(mob/living/carbon/user)
+	..()
```
Comment (silicons) [inline code/game/objects/items/granters.dm]: . = ..()
also why is oneuse specifically overriden here? is there no proc for this that you can override?

File: code/modules/antagonists/traitor/IAA/internal_affairs.dm [hunk 4]
```diff
@@ -226,18 +226,20 @@
...
+		to_chat(owner.current, "<span class='userdanger'>You have been provided with a standard uplink to accomplish your task. </span>")
+		to_chat(owner.current, "<span class='userdanger'>By no means reveal that you are a Syndicate agent. By no means reveal that your targets are being hunted.</span>")
 	else
-		to_chat(owner.current, "<span class='userdanger'>Your target is suspected of [crime], and you have been tasked with eliminating them by any means necessary to avoid a costly and embarrassing public trial.</span>")
-		to_chat(owner.current, "<B><font size=5 color=red>While you have a license to kill, unneeded property damage or loss of employee life will lead to your contract being terminated.</font></B>")
-		to_chat(owner.current, "<span class='userdanger'>For the sake of plausible deniability, you have been equipped with an array of captured Syndicate weaponry available via uplink.</span>")
+		to_chat(owner.current, "<span class='userdanger'>CAUTION: Your legal status as a citizen of NanoTrasen will be permanently revoked upon completion of your first contract.</span>")
```
Comment (silicons) [inline code/modules/antagonists/traitor/IAA/internal_affairs.dm:239]: players might get the wrong idea from this but whatever

File: code/game/objects/items/granters.dm (hunk 3 above)
Comment (Sishen1542) [reply code/game/objects/items/granters.dm]: `	var/oneuse = TRUE //default this is true, but admins can var this to 0 if we wanna all have a pass around of the rod form book`
All the other scrolls had it that way. Could probably refactor them all in the future.

File: code/modules/antagonists/traitor/IAA/internal_affairs.dm (hunk 4 above)
Comment (Sishen1542) [reply code/modules/antagonists/traitor/IAA/internal_affairs.dm:239]: Understandable, this is part of the memey Brigador reflavoring I introduced. Can always change it up

File: code/datums/components/uplink.dm (hunk 2 above)
Comment (Sishen1542) [reply code/datums/components/uplink.dm]: fixed.

**Comment by KeRSedChaplain** [comment]
bro I just realized that a new strat has become 
>thieving gloves
>find someone else speaking with codewords
>steal PDA/pen
>bam, gloves likely refunded + extra

**Pull request merged**